### PR TITLE
Feature/cbi 1321 record chainbridge transaction

### PR DESCRIPTION
**Pull request by karthikeyan-cere**
Extended the process extrinsic function to scan chain bridge functions

### CODE REVIEW

File: src/modules/block-scanner/block-scanner.service.ts [hunk 1]
```diff
@@ -496,6 +497,28 @@ export class BlockScannerService implements BlockScannerServiceInterface {
     }
   }
 
+  private processChainbridge(events: any): Promise<{ sender: string, args: string }> {
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.service.ts:500]: This method is synchronous. Remove promise.

File: src/modules/block-scanner/block-scanner.service.ts [hunk 2]
```diff
@@ -496,6 +497,28 @@ export class BlockScannerService implements BlockScannerServiceInterface {
...
+    return new Promise((resolve, reject) => {
+      events.forEach((value) => {
+        if (value.method === 'balances.Transfer') {
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.service.ts:503]: Move all method names to the constants.

File: src/modules/block-scanner/block-scanner.service.ts [hunk 3]
```diff
@@ -443,9 +445,22 @@ export class BlockScannerService implements BlockScannerServiceInterface {
         'staking.bond',
         'staking.validate',
         'staking.nominate',
+        'chainBridge.acknowledgeProposal',
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.service.ts:448]: Move all methods to the constants.

File: src/modules/block-scanner/block-scanner.service.ts (hunk 1 above)
Comment (AndreiNavoichyk) [reply src/modules/block-scanner/block-scanner.service.ts:500]: Rename to `extractSenderAndArgsFromChainbridge`

File: src/modules/block-scanner/block-scanner.service.ts [hunk 4]
```diff
@@ -443,9 +445,22 @@ export class BlockScannerService implements BlockScannerServiceInterface {
...
       ];
       extrinsic.forEach(async (txn, index) => {
         if (transferMethods.includes(txn.method)) {
+          if (txn.method === 'chainBridge.acknowledgeProposal') {
+            const tmpData : {sender: string, args: string} = await this.processChainbridge(txn.events);
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.service.ts:453]: rename to `extractedData`.

File: src/modules/block-scanner/block-scanner.service.ts [hunk 5]
```diff
@@ -496,6 +497,28 @@ export class BlockScannerService implements BlockScannerServiceInterface {
     }
   }
 
+  private processChainbridge(events: any): Promise<{ sender: string, args: string }> {
+    return new Promise((resolve, reject) => {
+      events.forEach((value) => {
+        if (value.method === 'balances.Transfer') {
+          const sender = value.data[0];
+          const args = `${value.data[1]}, ${value.data[2]}`;
+          resolve({sender, args});
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.service.ts:506]: `balance.Transfer` is not always on the list. This case should be handled.

File: src/modules/block-scanner/block-scanner.service.ts [hunk 6]
```diff
@@ -431,6 +431,8 @@ export class BlockScannerService implements BlockScannerServiceInterface {
   private processExtrinsics(extrinsic: any, block: any, network: string): any {
     try {
       const events = [];
+      let argument;
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.service.ts:434]: Is it 1 argument or can be a few? Rename to `arguments`.

Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts:434]: One argument.

File: src/modules/block-scanner/block-scanner.service.ts (hunk 4 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts:453]: renamed

File: src/modules/block-scanner/block-scanner.service.ts (hunk 1 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts:500]: rename. 
The function returns the Promise.

File: src/modules/block-scanner/block-scanner.service.ts (hunk 2 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts:503]: Moved

File: src/modules/block-scanner/block-scanner.service.ts (hunk 3 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts:448]: Moved

File: src/modules/block-scanner/block-scanner.service.ts (hunk 5 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts:507]: handled

File: src/modules/block-scanner/block-scanner.service.ts (hunk 1 above)
Comment (AndreiNavoichyk) [reply src/modules/block-scanner/block-scanner.service.ts]: You've added this function. It should not be asynchronous.

Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts]: Refactored.

File: src/modules/block-scanner/block-scanner.service.ts [hunk 7]
```diff
@@ -431,21 +432,39 @@ export class BlockScannerService implements BlockScannerServiceInterface {
...
+        MethodName.stakingBond,
+        MethodName.stakingValidate,
+        MethodName.stakingNominate,
+        MethodName.chainBridgeAckProposal,
       ];
       extrinsic.forEach(async (txn, index) => {
         if (transferMethods.includes(txn.method)) {
+          if (txn.method === MethodName.chainBridgeAckProposal) {
+            const extractedData: { sender: string, args: string } = this.extractSenderAndArgsFromChainbridge(txn.events);
+            if (extractedData === undefined) {
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.service.ts:455]: You should check here for null.

Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts:455]: updated.

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**